client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js

```diff
@@ -209,7 +209,7 @@ export const getCountriesData = ( state, orderId, siteId = getSelectedSiteId( st
 	};
 };
 
-const getAddressErrors = ( addressData, countriesData ) => {
+const getAddressErrors = ( addressData, countriesData, shouldValidatePhone = false ) => {
```

Comment:
> it would be nice to avoid passing boolean function arguments. they don't provide much clarity to the reader when they see them

I agree. I avoided passing a magic `true` argument in this case by using a vabiable: https://github.com/Automattic/wp-calypso/pull/24974/commits/600eebf0438fa94c91bbc69344f078e120fdd67a#diff-247e631a20d23f38190abb0dbe7f1223R445

> instead if we have parameters that control how a function executes we can pass a "bag of options" that gives the reader at the call-site more context

I'm not sure how I feel about the "bag of arguments" approach. It generally provides more context, but sometimes it can be taken too far. JavaScript already has a way of passing arguments to a function, which is positional arguments (if you use a half-decent editor, that will provide you context enough). I definitely see the value in using "bag of arguments" for flags like this, but in this case it's only 1 argument, and the function is just called from the same file. If it becomes more complicate, I'd refactor it to use bag of arguments or functional composition as you described.